src/Jira/Api.php

```diff
@@ -268,6 +288,43 @@ public function transition($issueKey, $params)
     }
 
     /**
+     * transation by step name
+     *
+     *
+     * @param $issueKey like YOURPROJ-22
+     * @param $stepName Step name like 'Done' or 'To Do'
+     * @param $params (array of parameters from JIRA API)
+     * @return mixed
+     */
+    public function transitionByStepName($issueKey, $stepName, $params = array())
+    {
+         $result = array();
+        //  get available transitions
+        $tmp_transitions = $this->getTransitions($issueKey, array());
+        $tmp_transitions_result = $tmp_transitions->getResult();
+        $transitions = $tmp_transitions_result['transitions'];
+
+        //  search id for closing ticket
+        foreach ($transitions as $v) {
```

Comment:
Boy scout rule: rename the `$v` variable to something more understandable (e.g. `$transitionData`).